src/plugins/data_view_editor/public/data_view_editor_service.ts

```diff
@@ -17,30 +25,117 @@ import {
 } from '@kbn/data-views-plugin/public';
 
 import { RollupIndicesCapsResponse, MatchedIndicesSet, TimestampOption } from './types';
-import { getMatchedIndices, ensureMinimumTime, extractTimeFields } from './lib';
+import { getMatchedIndices, ensureMinimumTime, extractTimeFields, removeSpaces } from './lib';
 import { GetFieldsOptions } from './shared_imports';
 
+export const matchedIndiciesDefault = {
+  allIndices: [],
+  exactMatchedIndices: [],
+  partialMatchedIndices: [],
+  visibleIndices: [],
+};
+
+export interface DataViewEditorServiceConstructorArgs {
+  services: {
+    http: HttpSetup;
+    dataViews: DataViewsServicePublic;
+  };
+  requireTimestampField?: boolean;
+  initialValues: {
+    name?: string;
+    type?: INDEX_PATTERN_TYPE;
+    indexPattern?: string;
+  };
+}
+
 export class DataViewEditorService {
-  constructor(private http: HttpSetup, private dataViews: DataViewsServicePublic) {
+  constructor({
+    services: { http, dataViews },
+    initialValues: {
+      type: initialType = INDEX_PATTERN_TYPE.DEFAULT,
+      indexPattern: initialIndexPattern = '',
+      name: initialName = '',
+    },
+    requireTimestampField = false,
+  }: DataViewEditorServiceConstructorArgs) {
```

Comment:
A redundant error message is being briefly shown when submitting a valid form.

For example after editing a time field:
![Nov-28-2022 10-27-33](https://user-images.githubusercontent.com/1415710/204242044-f80f1212-5f9a-4f7b-ac2d-faebbfdbe2d4.gif)

Reply:
addressed